core/capabilities/consensus/types/aggregator.go

```diff
@@ -0,0 +1,22 @@
+package types
+
+import (
+	ocrcommon "github.com/smartcontractkit/libocr/commontypes"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+type Aggregator interface {
+	// Called by the Outcome() phase of OCR reporting.
+	// The inner array of observations corresponds to elements listed in "inputs.observations" section.
+	Aggregate(previousOutcome *AggregationOutcome, observations map[ocrcommon.OracleID][]values.Value) (*AggregationOutcome, error)
```

Comment:
@bolekk How will the encoder know the underlying structure of the encodableOutcome (how to decode it)? I think that needs to be a Value for this to work.

Reply:
True, thanks! I think I can easily use a different proto type inside my AggregationOutcome.